cmd/kubectl-gadget/utils/k8s.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"context"
+
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/kubectl-gadget/configflags"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/k8sutil"
+)
+
+func GetRunningGadgetNamespace() (string, bool) {
```

Comment:
I'll make this function return an error in both cases: if there is not any or if there are more than 2 instances. I think it's better to fail explicitly rather than defaulting to the gadget namespace.

Reply:
Failing explicitly now excluding for `deploy`